tools/src/main/java/org/apache/kafka/trogdor/workload/ExternalCommandWorker.java

```diff
@@ -0,0 +1,259 @@
+                String line;
+                while ((line = br.readLine()) != null) {
+                    log.error("{}: (stderr of the process):{}", id, line);
+                    errMsg = line;
+                }
+            } catch (IOException ioe) {
+                log.info("{}: Stderr of the process is closed.", id);
+            } catch (Exception e) {
+                WorkerUtils.abort(log, "ErrorUpdater", e, doneFuture);
+            }
+        }
+    }
+
+    /**
+     * StatusUpdater reads ProcessWorker's stdout line by line. If the line is a JSON object, StatusUpdater updates
+     * the status with the JSON object.
+     */
+    public class StatusUpdater implements Runnable {
+        @Override
+        public void run() {
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8))) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    try {
+                        JsonNode resp = JsonUtil.JSON_SERDE.readTree(line);
+                        if (resp.has("status")) {
+                            status.update(resp.get("status"));
+                        }
+                        if (resp.has("log")) {
+                            log.info("{}: (stdout of the process): {}", id, resp.get("log").toString());
+                        }
+                        if (resp.has("error")) {
+                            JsonNode errNode = resp.get("error");
+                            if (errNode.getNodeType() == JsonNodeType.STRING) {
+                                errMsg = errNode.asText();
+                            }
+                        }
+                    } catch (IOException e) {
+                        // not a JSON string
+                    }
+                }
+            } catch (IOException ioe) {
+                log.info("{}: Stdout f the process is closed.", id);
+            } catch (Exception e) {
+                WorkerUtils.abort(log, "StatusUpdater", e, doneFuture);
+            }
+        }
+    }
+
+
+    @Override
+    public void stop(Platform platform) throws Exception {
+        if (!running.compareAndSet(true, false)) {
+            throw new IllegalStateException("ExternalCommandWorker is not running.");
+        }
+        log.info("{}: Deactivating ExternalCommandWorker.", id);
+        if (process != null) {
+            if (process.isAlive()) {
+                new ControlCommand("stop", spec).issue();
+            }
+            process.waitFor(1, TimeUnit.MINUTES);
+            if (process.isAlive()) {
+                log.info("{}: Destroy the external process since it is still alive after 1 minute.");
+                process.destroy();
+            }
+        }
+        doneFuture.complete(errMsg);
+        executor.shutdownNow();
+        executor.awaitTermination(1, TimeUnit.DAYS);
```

Comment:
Are we sure we want to wait at most 24hrs for this executor to fail?

Reply:
I copy that from ProduceBenchWorker, and it is same in ConsumeBenchWorker.
@cmccabe might know about the original motivation.